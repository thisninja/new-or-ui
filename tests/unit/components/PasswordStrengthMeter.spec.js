import Component from '@/components/PasswordStrengthMeter/main.vue';

test('renders correctly on weak passwords', () => {
  const { element } = shallowMount(Component, {
    propsData: {
      minEntropy: 80,
      value: 'password'
    }
  });
  expect(element).toMatchSnapshot();
});

test('renders correctly on strong passwords', () => {
  const { element } = shallowMount(Component, {
    propsData: {
      minEntropy: 80,
      value: 'XKCD.com/792/'
    }
  });
  expect(element).toMatchSnapshot();
});

test('renders correctly on password matcher', () => {
  const { element } = shallowMount(Component, {
    propsData: {
      match: 'password',
      passwordMatch: true,
      value: 'password'
    }
  });
  expect(element).toMatchSnapshot();
});
