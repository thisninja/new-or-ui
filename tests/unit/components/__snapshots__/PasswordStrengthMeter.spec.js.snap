// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly on password matcher 1`] = `
<div
  class="cl-password-strength-meter approved"
>
  <el-input
    placeholder=""
    type="password"
    value="password"
  >
    <i
      class="el-input__icon el-icon-view"
      slot="prefix"
    />
  </el-input>
   
  <!---->
</div>
`;

exports[`renders correctly on strong passwords 1`] = `
<div
  class="cl-password-strength-meter approved"
>
  <el-input
    placeholder=""
    type="password"
    value="XKCD.com/792/"
  >
    <i
      class="el-input__icon el-icon-view"
      slot="prefix"
    />
  </el-input>
   
  <div
    class="password-info"
  >
    <div
      class="entropy-container"
    >
      <div
        class="entropy-bar entropy-high"
        style="width: 53%; opacity: 0.53125;"
      >
        
                
            
      </div>
    </div>
     
    <span
      class="signal active"
    >
      
            abc
            
      <i
        class="el-icon-check active-marker"
      />
    </span>
    <span
      class="signal active"
    >
      
            ABC
            
      <i
        class="el-icon-check active-marker"
      />
    </span>
    <span
      class="signal active"
    >
      
            123
            
      <i
        class="el-icon-check active-marker"
      />
    </span>
    <span
      class="signal active"
    >
      
            !?#
            
      <i
        class="el-icon-check active-marker"
      />
    </span>
  </div>
</div>
`;

exports[`renders correctly on weak passwords 1`] = `
<div
  class="cl-password-strength-meter not-approved"
>
  <el-input
    placeholder=""
    type="password"
    value="password"
  >
    <i
      class="el-input__icon el-icon-view"
      slot="prefix"
    />
  </el-input>
   
  <div
    class="password-info"
  >
    <div
      class="entropy-container"
    >
      <div
        class="entropy-bar entropy-low"
        style="width: 23%; opacity: 0.76875;"
      >
        
                
            
      </div>
    </div>
     
    <span
      class="signal active"
    >
      
            abc
            
      <i
        class="el-icon-check active-marker"
      />
    </span>
    <span
      class="signal"
    >
      
            ABC
            
      <!---->
    </span>
    <span
      class="signal"
    >
      
            123
            
      <!---->
    </span>
    <span
      class="signal"
    >
      
            !?#
            
      <!---->
    </span>
  </div>
</div>
`;
